docs/content/posts/inductive_invariants.mdx

```diff
@@ -0,0 +1,511 @@
+              // update who to send to next
+              system' = system.set(p, { ... system.get(p), send_to: (system.get(p).send_to + 1) % (Proc.size() + 1)}),
+            }
+          | _ => all { // should not happen
+            network' = network,
+            system' = system
+          }
+        }
+     }
+```
+
+You now can run the REPL and investigate a number of runs to convince yourself that the model is doing what we described.
+
+#### Safety Property
+
+Let's now turn to analyzing the protocol with inductive invariants: The property safety is what we are looking for
+
+```quint
+val values_broadcast = network.filter(m => m.sender == initial_sender).map(m => m.payload)
+
+/// I only deliver messages that have been sent. This also means that if
+/// values_broadcast == Set() the no-one will ever deliver anything
+val safety =
+  Proc.forall(p =>
+    Domain.forall(v =>
+      system.get(p).s == Delivered(v) implies v.in(values_broadcast)))
+```
+
+#### TypeOK: the first constraint
+
+For using quint (and its back-end model checker Apalache) as the first step, you need to specify the state space in the following form. The name TypeOK is inherited from TLA+. It is jargon to specify the allowed content of state variables. This is a minimal requirement on what a correct state is:
+
+```quint
+val TypeOK = and {
+  network.in(msg_universe),
+  system.in(Proc.setOfMaps(state_space))
+}
+```
+
+`msg_universe` and `state_space` are variables that are defined over Cartesian products over all domains. Please refer to the quint spec for details. This is technically not so interesting,
+
+#### Deriving the Inductive Invariant
+
+So, no we specify the following candidate for our inductive invariant:
+
+```quint
+val IndInv = and {
+ TypeOK,
+ safety,
+}
+```
+
+...and with the following command asks quint two questions (1) is IndInv and inductive invariant and (2) if so, does it imply that safety holds. 
+
+```sh
+quint verify reliablebc.qnt --inductive-invariant IndInv --invariant safety
+```
+
+Here is what Quint has to say:
+
+```ansi /[State 0]/ /Init/1 /Send("goodbye")/
+[90m> [1/3] Checking whether the inductive invariant 'IndInv' holds in the initial state(s) defined by 'init'...[39m

+[90m> [2/3] Checking whether 'step' preserves the inductive invariant 'IndInv'...[39m

+[90mAn example execution:[39m

+[90m[39m

+[[1mState 0[22m]

+{

+  network[90m:[39m

+    [32mSet[39m(

+      { payload[90m:[39m [32m"goodbye"[39m, receiver[90m:[39m [33m2[39m, sender[90m:[39m [33m1[39m },

+      { payload[90m:[39m [32m"goodbye"[39m, receiver[90m:[39m [33m2[39m, sender[90m:[39m [33m2[39m }

+    ),

+  system[90m:[39m [32mMap[39m([33m1[39m[90m ->[39m { s[90m:[39m [32mInit[39m, send_to[90m:[39m [33m2[39m }, [33m2[39m[90m ->[39m { s[90m:[39m [32mSend[39m([32m"goodbye"[39m), send_to[90m:[39m [33m0[39m })


```

Comment:
Here the highlight is wrong. It should be on `{ payload: "goodbye", receiver: 2, sender: 1 }` instead of `Send("goodbye")`